web/source/osk/visualKeyboard.ts

```diff
@@ -701,77 +697,29 @@ namespace com.keyman.osk {
         return;
       }
 
-      // Do not move over keys if device popup visible
-      if(this.popupVisible) {
-        if(key1 == null) {
-          if(key0) {
-            this.highlightKey(key0,false);
-          }
-          this.keyPending=null;
-          this.touchPending=null;
-        } else {
-          if(key1 == this.popupBaseKey) {
-            if(!key1.classList.contains('kmw-key-touched')) {
-              this.highlightKey(key1,true);
-            }
-            this.keyPending = key1;
-            this.touchPending = e.touches[0];
-          } else {
-            if(key0) {
-              this.highlightKey(key0,false);
-            }
-            this.keyPending = null;
-            this.touchPending = null;
-          }
-        }
+      // Update all gesture tracking.  The function returns true if further input processing
+      // should be blocked.
+      if(this.updateGestures(key1, key0, e.changedTouches[0])) {
         return;
       }
 
-      var sk=document.getElementById('kmw-popup-keys');
-
-      // Use the popup duplicate of the base key if a phone with a visible popup array
-      if(sk && sk.style.visibility == 'visible' && this.device.formFactor == 'phone' && key1 == this.popupBaseKey) {
-        key1 = <KeyElement> sk.childNodes[0].firstChild;
-      }
-
       // Identify current touch position (to manage off-key release)
       this.currentTarget = key1;
 
-      // Clear previous key highlighting
-      if(key0 && key1 && key1 !== key0) {
-        this.highlightKey(key0,false);
-      }
-
-      // If popup is visible, need to move over popup, not over main keyboard
-      this.highlightSubKeys(key1,x,y);
-
-      if(sk && sk.style.visibility == 'visible') {
-        // Once a subkey array is displayed, do not allow changing the base key.
-        // Keep that array visible and accept no other options until the touch ends.
-        if(key1 && key1.id.indexOf('popup') < 0 && key1 != this.popupBaseKey) {
-          return;
-        }
+      // _Box has (most of) the useful client values.
+      let _Box = this.kbdDiv.parentElement ? this.kbdDiv.parentElement : keyman.osk._Box;
+      let height = this.kbdDiv.offsetHeight;
+      // We need to adjust the offset properties by any offsets related to the active banner.
 
-        // Highlight the base key on devices that do not append it to the subkey array.
-        if(key1 && key1 == this.popupBaseKey && key1.className.indexOf('kmw-key-touched') < 0) {
-          this.highlightKey(key1,true);
-        }
```

Comment:
Yeah, so 'in-browser' mode subkey highlighting and selection... was rather 'hacked in' with base-key highlighting and selection.  It's now _entirely_ handled within `SubkeyPopup.updateTouch`.